graylog2-server/src/main/java/org/graylog2/rest/resources/roles/RolesResource.java

```diff
@@ -0,0 +1,261 @@
+        }
+        return RoleResponse.create(roleToUpdate.getName(), Optional.fromNullable(roleToUpdate.getDescription()), roleToUpdate.getPermissions(),
+                                   role.readOnly());
+    }
+
+    @DELETE
+    @Path("{rolename}")
+    @ApiOperation(value = "Remove the named role")
+    public Response delete(@ApiParam(name = "rolename", required = true) @PathParam("rolename") String name) throws NotFoundException {
+        checkPermission(RestPermissions.ROLES_DELETE, name);
+
+        if (roleService.delete(name) == 0) {
+            throw new NotFoundException();
+        }
+        return Response.noContent().build();
+    }
+
+    @GET
+    @Path("{rolename}/members")
+    @RequiresPermissions({RestPermissions.USERS_LIST, RestPermissions.ROLES_READ})
+    @ApiOperation(value = "Retrieve the role's members")
+    public RoleMembershipResponse getMembers(@ApiParam(name = "rolename", required = true) @PathParam("rolename") String name) throws NotFoundException {
+        final Role role = roleService.load(name);
+
+        final Collection<User> users = userService.loadAllForRole(role);
+
+        Set<UserSummary> userSummaries = Sets.newHashSet();
+        for (User user : users) {
+            userSummaries.add(UserSummary.create(
+                    user.getId(),
+                    user.getName(),
+                    user.getEmail(),
+                    user.getFullName(),
+                    isPermitted(RestPermissions.USERS_PERMISSIONSEDIT,
+                                user.getName()) ? user.getPermissions() : Collections.<String>emptyList(),
+                    user.getPreferences(),
+                    firstNonNull(user.getTimeZone(), DateTimeZone.UTC).getID(),
+                    user.getSessionTimeoutMs(),
+                    user.isReadOnly(),
+                    user.isExternalUser(),
+                    user.getStartpage(),
+                    null
+            ));
+        }
+
+        return RoleMembershipResponse.create(role.getName(), userSummaries);
+    }
+
+    @PUT
+    @Path("{rolename}/members/{username}")
+    @ApiOperation("Add a user to a role")
+    public Response addMember(@ApiParam(name = "rolename") @PathParam("rolename") String rolename,
+                              @ApiParam(name = "username") @PathParam("username") String username,
+                              @ApiParam(name = "JSON Body", value = "Placeholder because PUT requests should have a body. Set to '{}', the content will be ignored.") String body) throws NotFoundException {
+        checkPermission(RestPermissions.ROLES_EDIT, username);
+
+        final User user = userService.load(username);
+        if (user == null) {
+            throw new NotFoundException("User " + username + " has not been found.");
+        }
+
+        // verify that the role exists
+        final Role role = roleService.load(rolename);
+
+        final HashSet<String> roles = Sets.newHashSet(user.getRoleIds());
```

Comment:
Unnecessary wrapping of `user.getRoleIds()` as its elements are being copied and transformed in `UserImpl#setRoleIds()`.